src/util/prolog/swiplwasmHandler.ts

```diff
@@ -0,0 +1,203 @@
+import fs from "fs";
+import path from "path";
+import { FS } from "../swipl-wasm/swipl-web";
+
+/**
+ * Runs a query on the consulted program.
+ * @param program The program to consult
+ * @returns A promise that is resolved when the new program is loaded.
+ */
+export function query(queryText: string): Promise<string> {
+	const previousPromise = lastPromise;
+	const promise = new Promise<string>(async (resolve, _) => {
+		let result: string;
+		await previousPromise;
+		printListener = str => {
+			result = str;
+			if (str.endsWith(".")) {
+				resolve(result);
+			} else {
+				// swipl-wasm cannot parse multiple results
+				// This line will display multiple results for a variable X in a bag
+				// Not a nice solution, could be improved upon. Works for diproche though.
+				return query("bagof(X, " + queryText.substr(0, queryText.length - 1) + ", Xs).");
```

Comment:
You could just do ``query = `bagof(X, ${queryText.substr(0, queryText.length - 1)}, Xs).`; `` in line 11. Then this function would always just behave the same and return a list of results (sometimes containing just a single element).

Reply:
This would cause issues with answers like "true" and only the same variable can be used which is X. But some more consistency really would be nice. I will see how I can manage this in the end.